pkg/crc/preflight/preflight_check_tray_darwin.go

```diff
@@ -103,11 +103,9 @@ func fixTrayExistsAndRunning() error {
 		return err
 	}
 	// get the tray app
-	if !trayAppCached() {
-		err := downloadOrExtractTrayApp()
-		if err != nil {
-			return err
-		}
+	err := downloadOrExtractTrayApp()
```

Comment:
This is added to `fixTrayExistsAndRunning()`, which only at `crc setup` time if `checkTrayExistsAndRunning` returned an error, so when is this expected to run exactly? Or does this assume the system tray has been stopped manually?
It would be better if we had a way of checking the version of the systray, and if the version is older than what the binary can install, then we upgrade. This is how it is done for the machine drivers.

Reply:
Now that the preflight check will fail if the tray is outdated, how does upgrade work exactly when the tray is running already? we'll install the new binary but won't restart the tray so that it's used, will we?

Reply:
@cfergeau Added a restart at the end of the fix function, thanks!